File: LeetCode_Marathons/LCM1/Easy/Is_Subsequence/a.go
func isSubsequence(s string, t string) bool {
    indexS, indexT := 0, 0

    for indexS != len(s) && indexT != len(t) {
        if (s[indexS] == t[indexT]) {
            indexS++
        }
        indexT++
    }

    return indexS == len(s)
}

File: LeetCode_Marathons/LCM1/Easy/Is_Subsequence/a.cpp
class Solution {
public:
    bool isSubsequence(string s, string t) {
        int indexS = 0, indexT = 0;

        while (indexS != s.size() && indexT != t.size()) {
            if (s[indexS] == t[indexT]) indexS++;
            indexT++;
        }

        return indexS == s.size();
    }
};